src/v/test_utils/http_imposter.cc

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "test_utils/http_imposter.h"
+
+#include "config/node_config.h"
+#include "vlog.h"
+
+#include <seastar/http/function_handlers.hh>
+
+#include <utility>
+
+static ss::logger http_imposter_log("http_imposter"); // NOLINT
+
+http_imposter_fixture::http_imposter_fixture() {
+    _server = ss::httpd::http_server_control();
+    _server.start().get();
+    ss::ipv4_addr ip_addr = {httpd_host_name.data(), httpd_port_number};
+    _server_addr = ss::socket_address(ip_addr);
+}
+
+http_imposter_fixture::~http_imposter_fixture() { _server.stop().get(); }
+
+const std::vector<ss::httpd::request>&
+http_imposter_fixture::get_requests() const {
+    return _requests;
+}
+
+const std::multimap<ss::sstring, ss::httpd::request>&
+http_imposter_fixture::get_targets() const {
+    return _targets;
+}
+
+void http_imposter_fixture::listen() {
+    _server.set_routes([this](ss::httpd::routes& r) { set_routes(r, _urls); })
+      .get();
+    _server.listen(_server_addr).get();
```

Comment:
what happens to exceptions thrown from handling requests?

Reply:
seastar seems to log them here https://github.com/scylladb/seastar/blob/master/src/http/httpd.cc#L517 from what I can see. Although it might be useful for us to manage this ourselves for app specific errors etc.